presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionExtractor.java

```diff
@@ -25,13 +27,22 @@
 import com.google.common.collect.ImmutableList;
 
 import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
 
 public class ExpressionExtractor
 {
     public static List<Expression> extractExpressions(PlanNode plan)
     {
+        return extractExpressions(plan, n -> n);
```

Comment:
`n -> n` is not sufficient Lookup stub here. With it, `ExpressionExtractor.Visitor#visitPlan` recurses till the end of the stack when it encounters `GroupReference`. https://github.com/prestodb/presto/pull/7982/commits/77ec5e9c5cacc7d41811be6f133b1b117d21fd7b would be more appropriate.